db/dbtypes/types.go

```diff
@@ -785,10 +785,10 @@ type AddressRowCompact struct {
 	Value          uint64
 }
 
-// AddressRowCompact is like AddressRow for efficient in-memory storage of
-// merged address transaction data. The fields are ordered to avoid unneeded
-// padding and extra data is omitted for efficient caching. The fields that only
-// pertain to non-merged views (IsFunding, TxVinVoutIndex, VinVoutDbID, and
+// AddressRowMerged is like AddressRow for efficient in-memory storage of merged
+// address transaction data. The fields are ordered to avoid unneeded padding
```

Comment:
The negation of the word `needed` sound a bit off even though its correct. IMO I feel like `unnecessary` would fit better.